### PR TITLE
certsyncpod: add reloader and fix starting informers

### PR DESCRIPTION
**Pull request by mfojtik**
Refactors the command to follow other commands structure.
Add file observer that kill the process when the kubeconfig file change.
Start the informer the controller need ;-)

### CODE REVIEW

File: pkg/operator/staticpod/certsyncpod/certsync_cmd.go [hunk 1]
```diff
@@ -55,6 +57,18 @@ func (o *CertSyncControllerOptions) Complete() (*CertSyncController, error) {
...
+	// When the kubeconfig content change, commit suicide to reload its content.
+	observer, err := fileobserver.NewObserver(500 * time.Second)
+	if err != nil {
+		return nil, err
+	}
+	initialContent, _ := ioutil.ReadFile(o.KubeConfigFile)
+	observer.AddReactor(fileobserver.ExitOnChangeReactor, map[string][]byte{o.KubeConfigFile: initialContent}, o.KubeConfigFile)
+
+	stopCh := make(chan struct{})
+	go observer.Run(stopCh)
```
Comment (sttts) [inline pkg/operator/staticpod/certsyncpod/certsync_cmd.go]: side effect in a Complete func looks fishy. Move it into a Start or Run method.

Comment (mfojtik) [reply pkg/operator/staticpod/certsyncpod/certsync_cmd.go]: sure

File: pkg/operator/staticpod/certsyncpod/certsync_cmd.go [hunk 2]
```diff
@@ -50,36 +55,65 @@ func NewCertSyncControllerCommand(configmaps, secrets []revision.RevisionResourc
...
 		o.Namespace,
 		o.configMaps,
 		o.secrets,
 		kubeInformers,
 		eventRecorder,
 	)
+	if err != nil {
+		return err
+	}
+	go controller.Run(1, stopCh)
```
Comment (sttts) [inline pkg/operator/staticpod/certsyncpod/certsync_cmd.go]: without go routine?

File: pkg/operator/staticpod/certsyncpod/certsync_cmd.go [hunk 3]
```diff
@@ -50,36 +55,65 @@ func NewCertSyncControllerCommand(configmaps, secrets []revision.RevisionResourc
...
-	// This kube client use protobuf, do not use it for CR
-	kubeClient, err := kubernetes.NewForConfig(protoKubeConfig)
-	if err != nil {
-		return nil, err
-	}
-	kubeInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, 10*time.Minute, informers.WithNamespace(o.Namespace))
+	initialContent, _ := ioutil.ReadFile(o.KubeConfigFile)
+	observer.AddReactor(fileobserver.ExitOnChangeReactor, map[string][]byte{o.KubeConfigFile: initialContent}, o.KubeConfigFile)
+
+	stopCh := make(chan struct{})
```
Comment (sttts) [inline pkg/operator/staticpod/certsyncpod/certsync_cmd.go]: why not move it into a param of `Run`, while touching this code.

Comment (mfojtik) [reply pkg/operator/staticpod/certsyncpod/certsync_cmd.go]: normally, CLI commands does not pass chans via Run(), just following conventions.

File: pkg/operator/staticpod/certsyncpod/certsync_cmd.go (hunk 2 above)
Comment (mfojtik) [reply pkg/operator/staticpod/certsyncpod/certsync_cmd.go]: i'm waiting for the stopCh to close, I think that is equal to what you suggests? (+I can print "this is shutting down)

Comment (sttts) [reply pkg/operator/staticpod/certsyncpod/certsync_cmd.go]: yes, it's the same. Just one line less :)

**Pull request merged**